File: CPP/Aksesoris.cpp
#include <bits/stdc++.h>
#include "Petshop.cpp"
using namespace std;

class Aksesoris : public Petshop {
    private:
        string jenis;
        string bahan;
        string warna;
    
    public:
        // Constructor Default
        Aksesoris() : Petshop() {
            jenis = "";
            bahan = "";
            warna = "";
        }
    
        // Constructor dengan parameter
        Aksesoris(string id, string nama, int stok, double harga, string jenis, string bahan, string warna)
            : Petshop(id, nama, stok, harga) {
            this->jenis = jenis;
            this->bahan = bahan;
            this->warna = warna;
        }
    
        // Getter dan Setter
        string getJenis() const { return jenis; }
        void setJenis(string jenis) { this->jenis = jenis; }
    
        string getBahan() const { return bahan; }
        void setBahan(string bahan) { this->bahan = bahan; }
    
        string getWarna() const { return warna; }
        void setWarna(string warna) { this->warna = warna; }
    };

File: PHP/Aksesoris.php
<?php
// Mengimpor class Petshop
include 'Petshop.php';

class Aksesoris extends Petshop {
    // Atribut tambahan untuk Aksesoris
    private $jenis = '';
    private $bahan = '';
    private $warna = '';

    // Constructor
    public function __construct($id = '', $nama = '', $stok = 0, $harga = 0.0, $foto = '', $jenis = '', $bahan = '', $warna = '') {
        // Memanggil constructor parent (Petshop)
        parent::__construct($id, $nama, $stok, $harga, $foto);
        $this->jenis = $jenis;
        $this->bahan = $bahan;
        $this->warna = $warna;
    }

    // Setter dan Getter untuk Aksesoris
    public function setJenis($jenis) {
        $this->jenis = $jenis;
    }

    public function getJenis() {
        return $this->jenis;
    }

    public function setBahan($bahan) {
        $this->bahan = $bahan;
    }

    public function getBahan() {
        return $this->bahan;
    }

    public function setWarna($warna) {
        $this->warna = $warna;
    }

    public function getWarna() {
        return $this->warna;
    }
}

?>
